app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    author.update(post_count: author.authored_posts.count)
+  end
+
+  # Method to return the 5 most recent comments for a given post
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  # Method to update the likes counter for a post
+  def update_likes_counter
+    update(likes_counter: likes.count)
+  end
```

Comment:
- [ ] The purpose of these counter functions is to increase the counter every time a like is saved in the database. Kindly make use of the model callback functions such as `after_save`. This function will save a counter every time a like is saved. 
__Kindly remove this function from here and take it into the likes model where it should belong__.

```ruby
  after_save :update_likes_counter
  private

  def update_llikes_counter
    author.increment!(:likes_counter)